cleanlab/object_detection/rank.py

```diff
@@ -0,0 +1,1055 @@
+    return wd
+
+
+# distance/similarity helped functions
+def _get_overlap(bb1: List[float], bb2: List[float]) -> float:
+    """Takes in two bounding boxes `bb1` and `bb2` and returns their IoU overlap."""
+
+    return _get_iou(_mod_coordinates(bb1), _mod_coordinates(bb2))
+
+
+def _get_overlap_matrix(bb1_list: np.ndarray, bb2_list: np.ndarray) -> np.ndarray:
+    """Takes in two lists of bounding boxes and returns an IoU matrix where IoU[i][j] is the overlap between
+    the i-th box in `bb1_list` and the j-th box in `bb2_list`."""
+    wd = np.zeros(shape=(len(bb1_list), len(bb2_list)))
+    for i in range(len(bb1_list)):
+        for j in range(len(bb2_list)):
+            wd[i][j] = _get_overlap(bb1_list[i], bb2_list[j])
+    return wd
+
+
+def _get_iou(bb1: Dict[str, Any], bb2: Dict[str, Any]) -> float:
+    """
+    Calculate the Intersection over Union (IoU) of two bounding boxes.
+    I've modified this to calculate overlap ratio in the line:
+    iou = np.clip(intersection_area / float(min(bb1_area,bb2_area)),0.0,1.0)
+
+    Parameters
+    ----------
+    bb1 : dict
+        Keys: {'x1', 'x2', 'y1', 'y2'}
+        The (x1, y1) position is at the top left corner,
+        the (x2, y2) position is at the bottom right corner
+    bb2 : dict
+        Keys: {'x1', 'x2', 'y1', 'y2'}
+        The (x, y) position is at the top left corner,
+        the (x2, y2) position is at the bottom right corner
+    Returns
+    -------
+    float
+        in [0, 1]
+    """
+    assert bb1["x1"] < bb1["x2"]
+    assert bb1["y1"] < bb1["y2"]
+    assert bb2["x1"] < bb2["x2"]
+    assert bb2["y1"] < bb2["y2"]
+
+    # determine the coordinates of the intersection rectangle
+    x_left = max(bb1["x1"], bb2["x1"])
+    y_top = max(bb1["y1"], bb2["y1"])
+    x_right = min(bb1["x2"], bb2["x2"])
+    y_bottom = min(bb1["y2"], bb2["y2"])
+
+    if x_right < x_left or y_bottom < y_top:
+        return 0.0
+
+    # The intersection of two axis-aligned bounding boxes is always an
+    # axis-aligned bounding box
+    intersection_area = (x_right - x_left) * (y_bottom - y_top)
+
+    # compute the area of both AABBs
+    bb1_area = (bb1["x2"] - bb1["x1"]) * (bb1["y2"] - bb1["y1"])
+    bb2_area = (bb2["x2"] - bb2["x1"]) * (bb2["y2"] - bb2["y1"])
+    assert intersection_area - 0.1 <= bb1_area
+    assert intersection_area - 0.1 <= bb2_area
+
+    # compute the intersection over union by taking the intersection
+    # area and dividing it by the sum of prediction + ground-truth
+    # areas - the interesection area
+    iou = intersection_area / float(bb1_area + bb2_area - intersection_area)
+    # There are some hyper-parameters here like consider tile area/object area
+    assert iou >= 0.0
+    assert iou - 0.01 <= 1.0
+    return iou
+
+
+def _euc_dis(box1: List[float], box2: List[float]) -> float:
+    """Calculates the euclidian distance between `box1` and `box2`."""
+    euc_factor = 0.1  # this is a hyperparameter
```

Comment:
move to constants.py as well, name in all caps, explain in comment (with explicit mention of object detection), etc. See my comments for other constants